artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -213,6 +230,15 @@ public OpenWireConnection(Connection connection,
       this.useKeepAlive = openWireProtocolManager.isUseKeepAlive();
       this.maxInactivityDuration = openWireProtocolManager.getMaxInactivityDuration();
       this.transportConnection.setProtocolConnection(this);
+      if (openWireProtocolManager.getMaxActorSize() > 0) {
+         this.maxActorSize = openWireProtocolManager.getMaxActorSize();
+      } else {
+         if (acceptorUsed == null) {
+            this.maxActorSize = TransportConstants.DEFAULT_TCP_RECEIVEBUFFER_SIZE;
+         } else {
+            this.maxActorSize = ((NettyAcceptor) acceptorUsed).getTcpReceiveBufferSize();
```

Comment:
this seems the only right option to me, if the actor can be used without an acceptor, then maybe pass in the maxPendingSize rather than the acceptorUsed.

Reply:
this will be null on tests only...